docs/node/migrations/overview.mdx

```diff
@@ -0,0 +1,26 @@
+---
+sidebar_position: 1
+sidebar_label: "Overview"
+id: migrations-overview
+title: Migrations
+description: Overview of Kwil's migration system
+slug: /node/migrations/overview
+---
+
+This section provides an overview of the migration system in Kwil. Migrations are used to upgrade the network to a new version or to a new network with minimal downtime. Depending on the nature of the upgrade, migrations can be coordinated or uncoordinated.
+
+## Rolling upgrades
+
+Whenever network has to be upgraded to a newer versions of `kwild` binaries which doesn't have any breaking changes to the consensus rules, the upgrade can be done in a rolling fashion. In a rolling upgrade, the validators can upgrade their nodes one by one without stopping the network and still ensure that the nodes agree on the app hash for the block.
+
+## Coordinated Upgrades
+
+Whenever there are breaking changes to the consensus rules the network has to be upgraded in a coordinated fashion. Kwil currently supports below two types of coordinated upgrades differing in the degree of coordination required and disruption tolerable during the upgrade process.
```

Comment:
Considering the terminology from https://github.com/cometbft/cometbft/wiki/Upgrades-Taxonomy, the two types of "migrations" we are documenting in this PR are the least favorable one where "entirely new chain is started with a different chain ID".  Their "coordinated upgrade" keeps the same chain, just coordinating the height at which the software needs to be updated.  We're starting a new chain in both.

> ## Network migration
> 
> When a software version includes breaking changes to the consensus rules, it may be necessary to migrate to a new network.  Kwil currently supports two types of migrations differing in the degree of coordination required and disruption that is tolerable.